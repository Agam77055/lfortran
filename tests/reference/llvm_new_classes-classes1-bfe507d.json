{
    "basename": "llvm_new_classes-classes1-bfe507d",
    "cmd": "lfortran --no-color --show-llvm --new-classes {infile} -o {outfile}",
    "infile": "tests/classes1.f90",
    "infile_hash": "2854104a42f1abb2d3573afedab08ac340e19b7d5151166ce3291182",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm_new_classes-classes1-bfe507d.stdout",
    "stdout_hash": "e551eeae505522ac05e23d2ca3da6ccd6be55723cbdc3b81e1257657",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}