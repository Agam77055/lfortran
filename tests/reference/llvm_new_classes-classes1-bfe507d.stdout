; ModuleID = 'LFortran'
source_filename = "LFortran"

%_VTable_base.type = type { [3 x i8*] }
%string_descriptor = type <{ i8*, i64 }>
%base = type <{ i32 (...)**, i32 }>

@_VTable_base = linkonce_odr unnamed_addr constant %_VTable_base.type { [3 x i8*] [i8* null, i8* null, i8* @__module_xx_show_x] }, align 8
@0 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@serialization_info = private unnamed_addr constant [3 x i8] c"I4\00", align 1
@1 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@string_const_data = private constant [11 x i8] c"ERROR STOP\00"
@string_const = private global %string_descriptor <{ i8* getelementptr inbounds ([11 x i8], [11 x i8]* @string_const_data, i32 0, i32 0), i64 10 }>
@string_const_data.1 = private constant [2 x i8] c"\0A\00"
@string_const.2 = private global %string_descriptor <{ i8* getelementptr inbounds ([2 x i8], [2 x i8]* @string_const_data.1, i32 0, i32 0), i64 1 }>
@2 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1

define void @__module_xx_show_x(%base* %this) {
.entry:
  %0 = getelementptr %base, %base* %this, i32 0, i32 1
  store i32 12, i32* %0, align 4
  br label %return

return:                                           ; preds = %.entry
  ret void
}

define i32 @main(i32 %0, i8** %1) {
.entry:
  call void @_lpython_call_initial_functions(i32 %0, i8** %1)
  %b = alloca %base, align 8
  %2 = bitcast %base* %b to i32 (...)***
  store i32 (...)** bitcast (i8** getelementptr inbounds (%_VTable_base.type, %_VTable_base.type* @_VTable_base, i32 0, i32 0, i32 2) to i32 (...)**), i32 (...)*** %2, align 8
  %3 = getelementptr %base, %base* %b, i32 0, i32 1
  %4 = getelementptr %base, %base* %b, i32 0, i32 1
  store i32 10, i32* %4, align 4
  call void @__module_xx_show_x(%base* %b)
  %5 = getelementptr %base, %base* %b, i32 0, i32 1
  %6 = load i32, i32* %5, align 4
  %7 = alloca i32, align 4
  store i32 %6, i32* %7, align 4
  %8 = call i8* (i8*, i64, i8*, i32, i32, ...) @_lcompilers_string_format_fortran(i8* null, i64 0, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @serialization_info, i32 0, i32 0), i32 0, i32 0, i32* %7)
  %9 = call i64 @_lfortran_str_len(i8* %8)
  %10 = call i8* @_lfortran_malloc(i64 16)
  %stringFormat_desc = bitcast i8* %10 to %string_descriptor*
  %11 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 0
  store i8* %8, i8** %11, align 8
  %12 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 1
  store i64 %9, i64* %12, align 4
  %13 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 0
  %14 = load i8*, i8** %13, align 8
  %15 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 1
  %16 = load i64, i64* %15, align 4
  %17 = trunc i64 %16 to i32
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @1, i32 0, i32 0), i8* %14, i32 %17, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @0, i32 0, i32 0), i32 1)
  %18 = getelementptr %base, %base* %b, i32 0, i32 1
  %19 = load i32, i32* %18, align 4
  %20 = icmp ne i32 %19, 12
  br i1 %20, label %then, label %else

then:                                             ; preds = %.entry
  %21 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const, i32 0, i32 0), align 8
  %22 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.2, i32 0, i32 0), align 8
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @2, i32 0, i32 0), i8* %21, i8* %22)
  call void @exit(i32 1)
  br label %ifcont

else:                                             ; preds = %.entry
  br label %ifcont

ifcont:                                           ; preds = %else, %then
  call void @_lpython_free_argv()
  br label %return

return:                                           ; preds = %ifcont
  ret i32 0
}

declare void @_lpython_call_initial_functions(i32, i8**)

declare i8* @_lcompilers_string_format_fortran(i8*, i64, i8*, i32, i32, ...)

declare i64 @_lfortran_str_len(i8*)

declare i8* @_lfortran_malloc(i64)

declare void @_lfortran_printf(i8*, i8*, i32, i8*, i32)

declare void @_lcompilers_print_error(i8*, ...)

declare void @exit(i32)

declare void @_lpython_free_argv()
